### PR TITLE
Add tests for IntervalsSourceProvider.Wildcard and IntervalsSourceProvider.Prefix

### PR DESCRIPTION
**Pull request by Hohol**
This PR adds unit tests for wire and xContent serialization of `IntervalsSourceProvider.Wildcard` and `IntervalsSourceProvider.Prefix`.

Relates https://github.com/elastic/elasticsearch/issues/50150

### CODE REVIEW

**Comment by Hohol** [comment]
@cbuescher I'm not sure what's the right workflow with dependent pull requests on GitHub. (And I wouldn't want to make them independent).
I guess it'll be inconvenient to review until the base PR is merged.
Let me know if you can suggest a better workflow.

File: server/src/test/java/org/elasticsearch/index/query/WildcardIntervalsSourceProviderTests.java [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+        return new Wildcard(
+            randomAlphaOfLength(10),
+            randomBoolean() ? randomAlphaOfLength(10) : null,
+            randomBoolean() ? randomAlphaOfLength(10) : null
+        );
+    }
+
+    @Override
+    protected Wildcard mutateInstance(Wildcard instance) throws IOException {
+        String Wildcard = instance.getPattern();
```
Comment (cbuescher) [inline server/src/test/java/org/elasticsearch/index/query/WildcardIntervalsSourceProviderTests.java]: nit: lowercase the variable name

File: server/src/test/java/org/elasticsearch/index/query/WildcardIntervalsSourceProviderTests.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    protected Writeable.Reader<Wildcard> instanceReader() {
+        return Wildcard::new;
+    }
+
+    @Override
+    protected Wildcard doParseInstance(XContentParser parser) throws IOException {
+        if (parser.nextToken() == XContentParser.Token.START_OBJECT) {
+            parser.nextToken();
+        }
+        Wildcard Wildcard = (Wildcard) IntervalsSourceProvider.fromXContent(parser);
```
Comment (cbuescher) [inline server/src/test/java/org/elasticsearch/index/query/WildcardIntervalsSourceProviderTests.java]: same here please

**Review by cbuescher — APPROVED** [review]
Hi @Hohol, looks great. I would prefer merging the two tests you already added now and continue with individual PRs for the others.
At least these two look independent to me. If you find writing the additional tests introduces dependencies you can of course do
all in one PR, I personally just find it harder to review larger chunks. The smaller the sooner you'll typically get feedback and get them merged.

Thanks a lot so far, I left two very minor code style issues, but the rest LGTM. I will wait with running the tests until you adressed those two small nitpicks.

**Comment by cbuescher** [comment]
@elasticmachine test this please

**Comment by cbuescher** [comment]
Looks like the bot didn't catch this.
@elasticmachine test this please

**Comment by Hohol** [comment]
What's wrong with elasticsearch-ci/1?

**Comment by Hohol** [comment]
Merged with master.

**Comment by cbuescher** [comment]
Lets try again
@elasticmachine test this please.

> What's wrong with elasticsearch-ci/1
Hard to say, the link to the build failure got lost after you merged another commit. Don't know how to get to it now but lets see what the next run brings.

**Comment by Hohol** [comment]
Failed again. Can you take a look at it?

**Comment by cbuescher** [comment]
The two new tests classes need the ES license headers, thats why some of the checks are failing atm. I think you can copy them over from any class or test class in the open source (everything under `server`).

**Comment by cbuescher** [comment]
@elasticmachine test this please

**Comment by cbuescher** [comment]
@Hohol great, thanks a lot, I merged to master and will also port to the 7.6 branch.

**Pull request merged**